web/src/emoji_picker.ts

```diff
@@ -535,11 +570,11 @@ function process_keypress(e) {
 
         if (pressed_key === 8) {
             // Handles Backspace.
-            new_query = old_query.slice(0, -1);
+            new_query = (typeof old_query === "string" ? old_query : "").slice(0, -1);
```

Comment:
This is not a safe way to convince TypeScript.
```
           const $emoji_filter = $<HTMLInputElement>(".emoji-popover-filter");
            const old_query = $emoji_filter.val()!;
```

Reply:
You might find [this](https://chat.zulip.org/#narrow/stream/6-frontend/topic/TypeScript.20migration/near/1750154) helpful